store/store_test.go

```diff
@@ -0,0 +1,336 @@
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/rand"
+	"go.uber.org/atomic"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/eds/edstest"
+	"github.com/celestiaorg/celestia-node/store/cache"
+)
+
+func TestEDSStore(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	t.Cleanup(cancel)
+
+	edsStore, err := NewStore(DefaultParameters(), t.TempDir())
+	require.NoError(t, err)
+
+	// disable cache
+	edsStore.cache = cache.NewDoubleCache(cache.NoopCache{}, cache.NoopCache{})
+	height := atomic.NewUint64(100)
+
+	// PutRegistersShard tests if Put registers the shard on the underlying DAGStore
+	t.Run("Put", func(t *testing.T) {
+		eds, dah := randomEDS(t)
+		height := height.Add(1)
+
+		f, err := edsStore.Put(ctx, dah.Hash(), height, eds)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+
+		// file should become available by hash
+		has, err := edsStore.HasByHash(ctx, dah.Hash())
+		require.NoError(t, err)
+		require.True(t, has)
+
+		// file should become available by height
+		has, err = edsStore.HasByHeight(ctx, height)
+		require.NoError(t, err)
+		require.True(t, has)
+	})
+
+	t.Run("Cached after Put", func(t *testing.T) {
+		edsStore, err := NewStore(DefaultParameters(), t.TempDir())
+		require.NoError(t, err)
+
+		eds, dah := randomEDS(t)
+		height := height.Add(1)
+
+		f, err := edsStore.Put(ctx, dah.Hash(), height, eds)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+
+		// file should be cached after put
+		f, err = edsStore.cache.Get(height)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+
+		// check that cached file is the same eds
+		fromFile, err := f.Shares(ctx)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+		expected := eds.FlattenedODS()
+		require.Equal(t, expected, fromFile)
+	})
+
+	t.Run("Second Put should be noop", func(t *testing.T) {
+		eds, dah := randomEDS(t)
+		height := height.Add(1)
+
+		f, err := edsStore.Put(ctx, dah.Hash(), height, eds)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+
+		f, err = edsStore.Put(ctx, dah.Hash(), height, eds)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+		// TODO: check amount of files in the store after the second Put
+		// after store supports listing
+	})
+
+	t.Run("GetByHeight", func(t *testing.T) {
+		eds, dah := randomEDS(t)
+		height := height.Add(1)
+
+		f, err := edsStore.Put(ctx, dah.Hash(), height, eds)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+
+		f, err = edsStore.GetByHeight(ctx, height)
+		require.NoError(t, err)
+
+		// check that cached file is the same eds
+		fromFile, err := f.Shares(ctx)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+		expected := eds.FlattenedODS()
+		require.Equal(t, expected, fromFile)
+	})
+
+	t.Run("GetByDataHash", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("GetByHash", func(t *testing.T) {
```

Ну либо метод поменять